socketio/src/socket.rs

```diff
@@ -322,90 +300,56 @@ impl Socket {
     /// This method is later registered as the callback for the `on_data` event of the
     /// engineio client.
     #[inline]
-    fn handle_new_message(&self, socket_bytes: Bytes) -> Option<SocketPacket> {
-        let mut is_finalized_packet = false;
-        // either this is a complete packet or the rest of a binary packet (as attachments are
-        // sent in a separate packet).
-        let decoded_packet = if self.unfinished_packet.read().unwrap().is_some() {
-            // this must be an attachement, so parse it
-            let mut unfinished_packet = self.unfinished_packet.write().unwrap();
-            let mut finalized_packet = unfinished_packet.take().unwrap();
-            finalized_packet.binary_data = Some(socket_bytes);
-
-            is_finalized_packet = true;
-            Ok(finalized_packet)
-        } else {
-            // this is a normal packet, so decode it
-            SocketPacket::try_from(&socket_bytes)
-        };
+    fn handle_new_packet(&self, socket_packet: SocketPacket) -> Option<SocketPacket> {
+        let output = socket_packet.clone();
 
-        if let Ok(socket_packet) = decoded_packet {
-            let output = socket_packet.clone();
+        let default = String::from("/");
+        //TODO: should nsp logic be here?
```

Comment:
Good question, where else does the packet get handled? Generally one could expect that such things happen in a method named `handle_new_packet`.

Reply:
Packet lifetime (planned for 0.3.0):
- SocketIO client socket connect method
  - SocketIO common socket connect method
  - Separate thread loop
    - SocketIO client socket iterator
      - Socketio common socket iterator
        - EngineIO iterator
          - Transport
          - EngineIO parsing
          - EngineIO handling (like responding to pings)
        - SocketIO common socket parsing
        - SocketIO common socket handling (no client side logic)
      - SocketIO client socket handling (client side logic)
Upcoming in future PRs

Reply:
Yep, lets handle this in a future PR ;)